pennylane_lightning/core/src/utils/cuda_utils/cuGates_host.hpp

```diff
@@ -1424,4 +1472,179 @@ static constexpr auto getP1111_CU() -> std::vector<CFP_t> {
             cuUtil::ONE<CFP_t>()};
 }
 
+template <class PrecisionT> class DynamicGateDataAccess {
```

Comment:
What is the purpose of this class? Can we have some guiding docs?

Reply:
Done!